apps/game_backend/lib/game_backend/quests/user_quest.ex

```diff
@@ -9,12 +9,18 @@ defmodule GameBackend.Quests.UserQuest do
   alias GameBackend.Quests.Quest
   alias GameBackend.Users.User
 
+  @derive {Jason.Encoder, only: [:id, :quest, :status, :progress, :goal, :activated_at]}
   schema "user_quests" do
     field(:completed_at, :utc_datetime)
     field(:status, :string)
+    field(:activated_at, :utc_datetime)
+
     belongs_to(:quest, Quest)
     belongs_to(:user, User)
 
+    field(:progress, :integer, virtual: true)
+    field(:goal, :integer, virtual: true)
```

Comment:
Why are these virtual?

Reply:
You don't want to save these fields in the db, you always want to calculate the progress of the quest.